components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1929,6 +1932,61 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         await this.userDB.deleteEnvVar(envvar);
     }
 
+    async hasSSHPublicKey(ctx: TraceContext): Promise<boolean> {
+        const user = this.checkUser("hasSSHPublicKey");
+        return this.userDB.hasSSHPublicKey(user.id);
+    }
+
+    async getSSHPublicKeys(ctx: TraceContext): Promise<UserSSHPublicKeyValue[]> {
+        const user = this.checkUser("getSSHPublicKeys");
+        const list = await this.userDB.getSSHPublicKeys(user.id);
+        return list.map((e) => ({
+            id: e.id,
+            name: e.name,
+            fingerprint: e.fingerprint,
+            creationTime: e.creationTime,
+            lastUsedTime: e.lastUsedTime,
+        }));
+    }
+
+    async addSSHPublicKey(ctx: TraceContext, value: SSHPublicKeyValue): Promise<UserSSHPublicKeyValue> {
+        const user = this.checkUser("addSSHPublicKey");
+        const data = await this.userDB.addSSHPublicKey(user.id, value);
+        this.updateSSHKeysForRegularRunningInstances(ctx, user.id);
```

Comment:
Here, you could show that you're intentionally not waiting for this to finish by adding a comment like so:

```suggestion
        /** no await */ this.updateSSHKeysForRegularRunningInstances(ctx, user.id);
```

Reply:
`tslint` print this error, so I change it to. [source](https://stackoverflow.com/a/52620074)
```ts
/** no await */ void this.updateSSHKeysForRegularRunningInstances(ctx, user.id);
```

![SCR-20220621-14p](https://user-images.githubusercontent.com/20944364/175326979-6def033d-9a4f-4f02-8336-5113f63f91b4.png)